### PR TITLE
move mplex implementation out of go-libp2p

### PR DESCRIPTION
**Pull request by sukunrt**

### CODE REVIEW

File: transport.go [hunk 1]
```diff
@@ -1,15 +1,28 @@
-// Deprecated: This package has moved into go-libp2p as a sub-package: github.com/libp2p/go-libp2p/p2p/muxer/mplex.
-package peerstream_multiplex
+package mplex
```
Comment (marten-seemann) [inline transport.go:1]: Should we keep the package-level deprecation?

**Review by marten-seemann — CHANGES_REQUESTED** [review]

File: transport.go (hunk 1 above)
Comment (sukunrt) [reply transport.go:3]: Yeah. We should.

Comment (sukunrt) [reply transport.go:3]: https://github.com/libp2p/go-libp2p-mplex/pull/38/commits/ccfff8377dedde91c6b74ed5e1db57e741471ee6

File: transport.go [hunk 2]
```diff
@@ -1,15 +1,30 @@
-// Deprecated: This package has moved into go-libp2p as a sub-package: github.com/libp2p/go-libp2p/p2p/muxer/mplex.
-package peerstream_multiplex
+// DEPRECATED: mplex has been deprecated. Users should prefer yamux over mplex. see https://github.com/libp2p/specs/issues/553
```
Comment (marten-seemann) [inline transport.go:1]:
```suggestion
// DEPRECATED: mplex has been deprecated. Users should prefer Yamux over mplex. see https://github.com/libp2p/specs/issues/553
```

**Comment by marten-seemann** [comment]
@sukunrt CI workflows were updated. Can you rebase please?

**Comment by sukunrt** [comment]
I missed this. I've rebased.

**Review by marten-seemann — APPROVED** [review]

**Pull request merged**